src/iqm/qiskit_iqm/iqm_naive_move_pass.py

```diff
@@ -66,6 +68,19 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         Raises:
             TranspilerError: The layout is not compatible with the DAG, or if the input gate set is incorrect.
         """
+        # pylint: disable=too-many-branches
+
+        # Replace symbolic parameters with indices and store the index to symbol mapping.
+        symbolic_gates = {}
+        symbolic_index = 0
+        for node in dag.topological_op_nodes():
+            # This only works for prx gates because that has two parameters
+            # We use one to mark that it is a symbolic gate (np.inf) and the other to store the index.
```

Comment:
Maybe worth adding a `TODO` or similar comment about the temporary nature of `inf`, and the fact that we should introduce an intermediate circuit representation to properly handle symbolic parameters.

Reply:
Done